### PR TITLE
Update extensions for the "web script" file associations

### PR DESCRIPTION
**Pull request by vlakoff**
Add `.mjs` extension to the "web script" file associations.

Related: #10746 (fixed by #12704).

Refs: [JavaScript modules#Aside — .mjs versus .js](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#aside_%E2%80%94_.mjs_versus_.js) on MDN

Side notes:
* `.jsm` seems to be a legacy extension for the same thing, dating back from when the module specs weren't fixed yet. I haven't looked into it further.
* `.jsp` is unrelated to JavaScript, it's [JavaServer Pages](https://en.wikipedia.org/wiki/Jakarta_Server_Pages) (basically Java embedded in HTML content, just like PHP embedded in HTML content in `.php` files).

### CODE REVIEW

**Comment by vlakoff** [comment]
Force-pushed to increase `nbExtMax` in order to fix this error:

> error C2078: too many initializers

**Comment by vlakoff** [comment]
Added a commit to remove the `.phtml` file extension.

As noted in this [answer on Stack Overflow](https://stackoverflow.com/questions/11859015/what-is-phtml-and-when-should-i-use-a-phtml-extension-rather-than-php/11859122#11859122):

> `.phtml` was the standard file extension for PHP 2 programs. `.php3` took over for PHP 3. When PHP 4 came out they switched to a straight `.php`.

As a reminder, `.phpN` file extensions are still used by some web hosting providers, to let the user indicate which PHP major version should execute the script.

File: PowerEditor/src/MISC/RegExt/regExtDlg.cpp [hunk 1]
```diff
@@ -48,8 +48,8 @@ const TCHAR defExtArray[nbSupportedLang][nbExtMax][extNameMax] =
 	{TEXT("web script"),
 		TEXT(".html"), TEXT(".htm"), TEXT(".shtml"), TEXT(".shtm"), TEXT(".hta"),
 		TEXT(".asp"), TEXT(".aspx"),
-		TEXT(".css"), TEXT(".js"), TEXT(".json"), TEXT(".jsm"), TEXT(".jsp"),
-		TEXT(".php"), TEXT(".php3"), TEXT(".php4"), TEXT(".php5"), TEXT(".phps"), TEXT(".phpt"), TEXT(".phtml"),
```
Comment (donho) [inline PowerEditor/src/MISC/RegExt/regExtDlg.cpp:52]: .phtml "was" the standard file extension for PHP 2 programs, doesn't mean it doesn't exist anymore.
I could imagine there are still some servers use php2 so some phtml files still exist. 
OTOH, there's no reason to remove it if we keep php3 & php4 which are taken over by php5.

**Review by donho — CHANGES_REQUESTED** [review]
Restore .phtml please.

File: PowerEditor/src/MISC/RegExt/regExtDlg.cpp (hunk 1 above)
Comment (vlakoff) [reply PowerEditor/src/MISC/RegExt/regExtDlg.cpp:52]: Come on, PHP 2 has been released on 1997! It's so old it's not even listed on the [Unsupported Historical Releases
](https://www.php.net/releases/index.php) page lol.

Here I'm just applying the [Less is more](https://en.wikipedia.org/wiki/Less_is_more) principle, removing a tiny bit of low-hanging bloat, to improves usability of the relevant parts.

Comment (donho) [reply PowerEditor/src/MISC/RegExt/regExtDlg.cpp:52]: My point is phtml could still be used.

**Comment by donho** [comment]
Create a new PR if you want to add only .jsm

**Comment by vlakoff** [comment]
New PR: #14689

**Pull request closed without merging**